### DISCUSSION TITLE
Is editing strings in decompilation of binarys a planned feature?

### DISCUSSION THREAD

**Discussion by SirEnder125** [NationalSecurityAgency/ghidra#2692] [Ideas]
Hello. I want to edit what a certain program writes to the console, is this a planned feature??

**Comment by dragonmacher** [comment]
Can you elaborate on your question?   Do you mean you wish to edit the binary that you have open in Ghidra such that you can change it's bytes to change its behavior when you run that binary?   Or, do you mean you wish to have Ghidra send messages to the Ghidra or system console?

**Reply by SirEnder125** [reply]
All I mean is that I want to edit some of the strings in the program. If the program, for instance used a function such as
'printf' or 'std::cout <<' I could edit what messages it wrote. I found no way to do this, is it possible?

**Reply by dragonmacher** [reply]
That would require changing the bytes of the program.  While changing bytes is possible in Ghidra, the tool is not meant for patching programs.

**Reply by SirEnder125** [reply]
Okay, thanks. I just edit them and save?

**Reply by dragonmacher** [reply]
You can edit bytes in the Byte Viewer.   You can also patch instructions, which is done from the Listing (see the help for patching).   Save any changes you wish to keep.     The issue you will run into is that Ghidra does not allow for the export of the binary back into an executable form.   That is a feature that we do not support.
